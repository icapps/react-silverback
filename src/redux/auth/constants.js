export default {
  LOGIN_USER: 'LOGIN_USER',
  LOGIN_USER_PENDING: 'LOGIN_USER_PENDING',
  LOGIN_USER_REJECTED: 'LOGIN_USER_REJECTED',
  LOGIN_USER_FULFILLED: 'LOGIN_USER_FULFILLED',
  LOGOUT_USER: 'LOGOUT_USER',
};
