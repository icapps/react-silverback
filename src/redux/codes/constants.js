export default {
  GET_LANGUAGE_CODES: 'GET_LANGUAGE_CODES',
  GET_LANGUAGE_CODES_PENDING: 'GET_LANGUAGE_CODES_PENDING',
  GET_LANGUAGE_CODES_FULFILLED: 'GET_LANGUAGE_CODES_FULFILLED',
  GET_LANGUAGE_CODES_REJECTED: 'GET_LANGUAGE_CODES_REJECTED',
};
