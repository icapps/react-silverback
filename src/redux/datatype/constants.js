export default {
  GET_DATA_TYPE: 'GET_DATA_TYPE',
  GET_DATA_TYPE_PENDING: 'GET_DATA_TYPE_PENDING',
  GET_DATA_TYPE_REJECTED: 'GET_DATA_TYPE_REJECTED',
  GET_DATA_TYPE_FULFILLED: 'GET_DATA_TYPE_FULFILLED',
};
