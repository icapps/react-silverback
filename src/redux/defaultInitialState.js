const defaultInitialState = {
  isPending: false,
  isError: false,
  errorMessage: '',
};

export default defaultInitialState;
