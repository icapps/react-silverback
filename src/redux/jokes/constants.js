export default {
  GET_JOKE           : 'GET_JOKE',
  GET_JOKE_REQUEST   : 'GET_JOKE_REQUEST',
  GET_JOKE_FULFILLED : 'GET_JOKE_FULFILLED',
  GET_JOKE_REJECTED  : 'GET_JOKE_REJECTED',
};
