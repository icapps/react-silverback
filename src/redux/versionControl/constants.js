export default {
  GET_VERSION: 'GET_VERSION',
  GET_VERSION_PENDING: 'GET_VERSION_PENDING',
  GET_VERSION_FULFILLED: 'GET_VERSION_FULFILLED',
  GET_VERSION_REJECTED: 'GET_VERSION_REJECTED',
};
