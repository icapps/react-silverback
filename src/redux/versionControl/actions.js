import constants from './constants';
import { versionControlService } from '../../services';

export const getVersion = () => {
  return {
    type: constants.GET_VERSION,
    payload: versionControlService.get(),
  };
};
