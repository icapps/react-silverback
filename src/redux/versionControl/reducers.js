import constants from './constants';
import defaultInitialState from '../defaultInitialState';
import { strings } from '../../utils/index';

const initialState = {
  build: '',
  version: '',
  ...defaultInitialState,
};

const errorMessageHandling = payload => payload.errors[0].detail || payload.errors[0].title || strings.GENERAL_ERROR;

const versionControl = (state = initialState, action = {}) => {
  const { payload } = action;

  switch (action.type) {
    case constants.GET_VERSION_FULFILLED:
      return {
        ...state,
        build: payload.data.build,
        version: payload.data.version,
        isPending: false,
      };
    case constants.GET_VERSION_PENDING:
      return {
        ...state,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.GET_VERSION_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: errorMessageHandling(payload),
      };
    default:
      return state;
  }
};

export default versionControl;
