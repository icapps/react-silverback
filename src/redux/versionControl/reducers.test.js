import versionControl from './reducers';
import constants from './constants';
import defaultInitialState from '../defaultInitialState';

const initialState = {
  build: '',
  version: '',
  ...defaultInitialState,
};

describe('versionControl reducer', () => {
  it('should return the initial state', () => {
    expect(versionControl(undefined, {})).toMatchSnapshot();
  });
  it('should GET_VERSION_FULFILLED', () => {
    expect(
      versionControl(initialState, {
        type: constants.GET_VERSION_FULFILLED,
        payload: { data: { build: 'v51', version: '1.0.1' }, meta: { totalCount: 10 } },
      })
    ).toMatchSnapshot();
  });
  it('should GET_VERSION_PENDING', () => {
    expect(
      versionControl(initialState, {
        type: constants.GET_VERSION_PENDING,
      })
    ).toMatchSnapshot();
  });
  it('should GET_VERSION_REJECTED', () => {
    expect(
      versionControl(initialState, {
        type: constants.GET_VERSION_REJECTED,
        payload: { errors: [{ detail: 'ErrorMessage' }] },
      })
    ).toMatchSnapshot();
  });
});
