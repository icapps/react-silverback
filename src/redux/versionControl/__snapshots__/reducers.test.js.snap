// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`versionControl reducer should GET_VERSION_FULFILLED 1`] = `
Object {
  "build": "v51",
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "version": "1.0.1",
}
`;

exports[`versionControl reducer should GET_VERSION_PENDING 1`] = `
Object {
  "build": "",
  "errorMessage": "",
  "isError": false,
  "isPending": true,
  "version": "",
}
`;

exports[`versionControl reducer should GET_VERSION_REJECTED 1`] = `
Object {
  "build": "",
  "errorMessage": "ErrorMessage",
  "isError": true,
  "isPending": false,
  "version": "",
}
`;

exports[`versionControl reducer should return the initial state 1`] = `
Object {
  "build": "",
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "version": "",
}
`;
